### PR TITLE
feat: Limit scope of views

### PR DESCRIPTION
**Pull request by shimonp21**

### CODE REVIEW

File: pkg/policy/execute.go [hunk 1]
```diff
@@ -158,6 +155,11 @@ func (e *Executor) Execute(ctx context.Context, req *ExecuteRequest, policy *Pol
 		}
 	}
 
+	if err := e.createViews(ctx, policy); err != nil {
+		return nil, err
+	}
+	defer e.deleteViews(ctx, policy)
+
```
Comment (roneli) [inline pkg/policy/execute.go:162]: In theory the better design here is to create a session at this point (pool.Acquire) run the checks and close the session rather then create delete. The original passing of only a connection is a flaw by design. (I am not sure if its a simple fix as this, so if it isn't maybe add a todo here).

**Review by roneli — COMMENTED** [review]
Had one comment, once it's resolved either with todo or the actual fix will approve and we can merge.

File: pkg/policy/execute.go (hunk 1 above)
Comment (shimonp21) [reply pkg/policy/execute.go:164]: Added TODO for now. Definitely agree that a new session per policy is a better idea, but this refactor is not trivial.

**Review by roneli — APPROVED** [review]
LGTM

**Pull request merged**